src/routes/userRouter.ts

```diff
@@ -12,4 +12,7 @@ router.get("/admin-get-user/:id",userAuthorization(["admin"]), isUserExist, user
 router.put("/admin-update-user-status/:id",userAuthorization(["admin"]), validation(statusSchema), isUserExist, userControllers.updateUserStatus);
 router.put("/admin-update-role/:id",userAuthorization(["admin"]), validation(roleSchema),isUserExist, userControllers.updateUserRole);
 
+router.get("/user-get-profile/:id",userAuthorization(["buyer", "seller"]),userControllers.getUserDetails)
```

Comment:
We don't have to parse user id into parameters as we are getting a user id from userAuthorization middleware.